tests/components/blebox/test_init.py

```diff
@@ -0,0 +1,63 @@
+"""BleBox devices setup tests."""
+
+import blebox_uniapi
+import pytest
+
+from homeassistant.components.blebox import async_setup_entry
+from homeassistant.components.blebox.const import DOMAIN
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .conftest import mock_config, patch_product_identify
+
+from tests.async_mock import ANY, call, patch
+
+
+async def test_setup_failure(hass):
+    """Test that setup failure is handled and logged."""
+
+    patch_product_identify(None, side_effect=blebox_uniapi.error.ClientError)
+
+    with patch("homeassistant.components.blebox._LOGGER.error") as error:
+        with pytest.raises(ConfigEntryNotReady):
+            config = mock_config()
+            config.add_to_hass(hass)
+            await async_setup_entry(hass, config)
```

Comment:
Please use the core interface `hass.config_entries.async_setup` to set up the config entry.